### DISCUSSION TITLE
Research what trends and patterns can be identify using user's income data #125

### DISCUSSION THREAD

**Discussion by Fernando2908** [uprm-inso4101-2023-2024-s2/semester-project-personalfinancemanager#133] [Ideas]
### **_Seasonal Variations:_**

- Investigate whether there are seasonal patterns in income.

- Some industries or professions may experience fluctuations during specific periods of the year.

### **_Income Growth:_**

- Monitor the growth or decline of income over time.

- Explore the factors contributing to changes in income.

### **_Income Stability:_**

- Evaluate the stability of income sources.

- Identify any patterns related to job changes, promotions, or periods of unemployment.

### **Predictive Analysis:**

- Examine past income patterns to build models that anticipate future income. This facilitates financial planning and forecasting future needs.

### **_Income Source Analysis:_**

- Identify the main sources from which users derive the majority of their income.

- Assess the frequency and volume of income from these sources over time.

- Use this information to understand user preferences and behaviors.

**Comment by JeanMontesSantiago** [comment]
Hello Fernando,
I like the ideas of "Income Source Analysis" and "Income Growth." I also think that implementing these ideas might be challenging. How do you plan to implement them? Lastly, we could create a section dedicated to calculating the average, mean, median, and standard deviation of income.

**Comment by Fernando2908** [comment]
## **Implementation Ideas:**

Using these ideas, we can implement both "Income Source Analysis" and "Income Growth."

**Income Categorization by Source:**

- Provide users with the ability to categorize each transaction by entering the income source (salary, investments, freelance, etc.).

**Sorting Categories from Highest to Lowest Income:**

- Implement a function that calculates the total income for each income source.
- Sort the income sources from highest to lowest based on their total income.

**Category Details:**

- Allow users to click on a specific income source to get more details about the transactions associated with that source.

**Temporal Analysis of Income Sources:**

- Implement an income history for each category that shows how the contribution of each income source has evolved over time.

**Key Metrics:**

- Calculate and display key metrics such as average monthly income, median, and standard deviation for each income source.
